pkg/sql/executor.go

```diff
@@ -1098,8 +1095,9 @@ func (e *Executor) execStmtsInCurrentTxn(
 		stmt.queryID = queryID
 		stmt.queryMeta = queryMeta
 
-		// Fork context for this statement.
-		queryMeta.ctx, queryMeta.cancel = context.WithCancel(session.Ctx())
+		// TODO(itsbilal): Fork a statement-specific context here, that gets cancelled
+		// upon request by user.
```

Comment:
As mentioned privately, perhaps we should create a new context whenever a transaction starts.